File: IoWarningEventArgs.cs


using System;

namespace kCura.WinEDDS.TApi
{
    /// <summary>
    /// Class for IO warning event arguments
    /// </summary>
    public class IoWarningEventArgs : EventArgs
    {
        /// <summary>
        /// IoWarningEventArgs exception types
        /// </summary>
        public enum TypeEnum
        {
            /// <summary>
            /// Message
            /// </summary>
            Message,
            /// <summary>
            /// Instant Retry Error
            /// </summary>
            InstantRetryError,
            /// <summary>
            /// Wait Retry Error
            /// </summary>
            WaitRetryError
        }

        private int _waitTime;
        private System.Exception _exception;
        private string _message;
        private long _currentLineNumber;

        private TypeEnum _type;

        /// <summary>
        /// Contructor for IoWarningEventArgs
        /// </summary>
        public IoWarningEventArgs(int waitTime, System.Exception ex, string message, long currentLineNumber)
        {
            _waitTime = waitTime;
            _exception = ex;
            _currentLineNumber = currentLineNumber;
            _message = message;
            if (waitTime > 0)
            {
                _type = TypeEnum.WaitRetryError;
            }
            else
            {
                _type = TypeEnum.InstantRetryError;
            }
        }

        /// <summary>
        /// Contructor for IoWarningEventArgs
        /// </summary>
        public IoWarningEventArgs(string message, long currentLineNumber)
        {
            _message = message;
            _currentLineNumber = currentLineNumber;
            _type = TypeEnum.Message;
        }

        /// <summary>
        /// Wait time
        /// </summary>
        public int WaitTime => _waitTime;

        /// <summary>
        /// Exception
        /// </summary>
        public System.Exception Exception => _exception;

        /// <summary>
        /// Current line number
        /// </summary>
        public long CurrentLineNumber => _currentLineNumber;

        /// <summary>
        /// Message
        /// </summary>
        public string Message => _message;

        /// <summary>
        /// Type
        /// </summary>
        public TypeEnum Type => _type;
    }
}

File: Source/Relativity.Import.Export.Legacy/API/IoWarningEventArgs.vb
Namespace kCura.WinEDDS.Api
	Public Class IoWarningEventArgs
		Public Enum TypeEnum
			Message
			InstantRetryError
			WaitRetryError
		End Enum

		Private _waitTime As Int32
		Private _exception As System.Exception
		Private _currentLineNumber As Int64
		Private _message As String
		Private _type As TypeEnum

		Public Sub New(ByVal waitTime As Int32, ByVal ex As System.Exception, ByVal currentLineNumber As Int64)
			_waitTime = waitTime
			_exception = ex
			_currentLineNumber = currentLineNumber
			If waitTime > 0 Then
				_type = TypeEnum.WaitRetryError
			Else
				_type = TypeEnum.InstantRetryError
			End If
		End Sub
		Public Sub New(ByVal message As String, ByVal currentLineNumber As Int64)
			_message = message
			_currentLineNumber = currentLineNumber
			_type = TypeEnum.Message
		End Sub

		Public ReadOnly Property WaitTime() As Int32
			Get
				Return _waitTime
			End Get
		End Property
		Public ReadOnly Property Exception() As System.Exception
			Get
				Return _exception
			End Get
		End Property
		Public ReadOnly Property CurrentLineNumber() As Int64
			Get
				Return _currentLineNumber
			End Get
		End Property
		Public ReadOnly Property Message() As String
			Get
				Return _message
			End Get
		End Property

		Public ReadOnly Property Type() As TypeEnum
			Get
				Return _type
			End Get
		End Property

	End Class
End Namespace

